MultimodalQnA/ui/gradio/multimodalqna_ui_gradio.py

```diff
@@ -315,6 +315,64 @@ def ingest_image_gen_caption(filepath, request: gr.Request):
     return
 
 
+def ingest_image_with_text(filepath, text, request: gr.Request):
+    yield (gr.Textbox(visible=True, value="Please wait for your uploaded image to be ingested into the database..."))
+    verified_filepath = os.path.normpath(filepath)
+    if not verified_filepath.startswith(tmp_upload_folder):
+        print("Found malicious image file name!")
+        yield (
+            gr.Textbox(
+                visible=True,
+                value="Your uploaded image's file name has special characters that are not allowed. Please consider updating the file name!",
+            )
+        )
+        return
+    basename = os.path.basename(verified_filepath)
+    dest = os.path.join(static_dir, basename)
+    shutil.copy(verified_filepath, dest)
+    text_basename = "{}.txt".format(os.path.splitext(basename)[0])
+    text_dest = os.path.join(static_dir, text_basename)
+    with open(text_dest, "a") as file:
```

Comment:
What's the reason for appending versus write? Could that potentially mean we are getting text from last time? Should be be checking if the file exists first?

Reply:
I'll change it to`w`. I think that's better.